### PR TITLE
feat:Setting of Routes Skeletons (REST API) (#39523)

### PR DESCRIPTION
**Pull request by Ankush-Negi**

### CODE REVIEW

File: src/Server.ts [hunk 1]
```diff
@@ -1,37 +1,49 @@
 import IConfig from './config/IConfig';
 import * as express from 'express';
+import * as bodyParser from 'body-parser';
+import errHandler from './libs/routes/errorHandler';
+import notFoundRoute from './libs/routes/notFoundRoute';
+import {default as router} from './router';
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: don't use default while importing

File: src/Server.ts [hunk 2]
```diff
@@ -1,37 +1,49 @@
...
+        return this;
     }
 
     setupRoutes = () => {
         this.app.use('/health-check', ( req, res) => {
             res.send('I am OK.');
-            return this;
         });
-    }
+        this.app.use('/api', router);
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: destructure this.app.use

File: src/controllers/trainee/Controller.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            status: 'OK',
+            message: 'trainee updated successfully',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
+            }
+        });
+    }
+    findAll = (req:Request, res:Response) => {
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: name this as getAll

File: src/controllers/trainee/Controller.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            status: 'OK',
+            message: 'trainee list is: \n',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
+            }
+        });
+    }
+    findOne = (req:Request, res:Response) => {
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: rename this as getById

File: src/controllers/trainee/Controller.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    create = (req: Request,res:Response)=>{
+        res.send({
+            status: 'OK',
+            message: 'trainee added successfully',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
+            }
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: keep this data outside in separate object and pass that in response

File: src/controllers/trainee/Controller.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    static getInstance = () => {
+    if(!TraineeController.instance)
+    {
+        TraineeController.instance= new TraineeController();
+
+    }
+    return TraineeController.instance;
+    }
+    create = (req: Request,res:Response)=>{
+        res.send({
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: pass the request through the body. 
along with logging the request 
        const { name, id } = req.body;

Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: add custom validation here by checking the required fields

File: src/controllers/trainee/Controller.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            status: 'OK',
+            message: 'trainee added successfully',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
+            }
+        });
+    }
+    delete = (req:Request, res:Response) => {
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts:31]: pass the req through params console the req.params passed

File: src/controllers/trainee/Controller.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        res.send({
+            status: 'OK',
+            message: 'trainee deleted successfully',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
+            }
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: data to be kept outside in an obj

File: src/controllers/trainee/Controller.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        });
+    }
+    update = (req:Request, res:Response) => {
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts:48]: - req must be sent through body
- log the body
- keep data in separate obj

File: src/controllers/trainee/Controller.ts [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        res.send({
+            status: 'OK',
+            message: 'trainee updated successfully',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: separate this data

File: src/controllers/trainee/Controller.ts [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    findAll = (req:Request, res:Response) => {
+        console.log(':::::::::::Inside FindAll trainee::::::::::::::');
+        res.send({
+            status: 'OK',
+            message: 'trainee list is: \n',
+            data:{
+                id: 1,
+                name: 'Ankush',
+                address: 'Delhi'
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/Controller.ts]: separate this data

File: src/controllers/trainee/routes.ts [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import TraineeController from './Controller';
+
+const traineeRouter = Router();
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/routes.ts]: define the type of trainee router here

File: src/controllers/user/Controller.ts [hunk 13]
```diff
@@ -0,0 +1,70 @@
+import {Request, Response} from 'express';
+class UserController{
```
Comment (VisaaliSuccessive) [inline src/controllers/user/Controller.ts:3]: all same mistakes done in trainee module to be corrected here

File: src/controllers/user/routes.ts [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import UserController from './Controller';
+
+const userRouter = Router();
```
Comment (VisaaliSuccessive) [inline src/controllers/user/routes.ts]: define the type of userRouter

File: src/Server.ts (hunk 1 above)
Comment (Ankush-Negi) [reply src/Server.ts]: Done

File: src/Server.ts (hunk 2 above)
Comment (Ankush-Negi) [reply src/Server.ts]: Done

File: src/controllers/trainee/Controller.ts (hunk 3 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/Controller.ts]: Done

File: src/controllers/trainee/Controller.ts (hunk 4 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/Controller.ts]: Done

File: src/controllers/trainee/Controller.ts (hunk 5 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/Controller.ts]: Done

File: src/controllers/trainee/Controller.ts (hunk 10 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/Controller.ts]: Done

File: src/controllers/trainee/Controller.ts (hunk 11 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/Controller.ts]: Done

File: src/controllers/trainee/routes.ts (hunk 12 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/routes.ts]: Done

File: src/controllers/user/Controller.ts (hunk 13 above)
Comment (Ankush-Negi) [reply src/controllers/user/Controller.ts:3]: Done

File: src/controllers/user/routes.ts (hunk 14 above)
Comment (Ankush-Negi) [reply src/controllers/user/routes.ts]: Done

**Review by VisaaliSuccessive — APPROVED** [review]

**Pull request merged**